### ISSUE TITLE
feat: Remove generic hasher from block hash computation

### ISSUE THREAD

**Issue by tdelabro** [keep-starknet-strange/madara#1572]
# Feature Request

Starknet BlockHash computation should always use PedersenHasher

**Describe Preferred Solution**

In `crates/primitives/block/src/header.rs` replaces `Header::hash` by
```rust
    pub fn hash(&self) -> Felt252Wrapper {
        let data: &[Felt252Wrapper] = &[
            self.block_number.into(),
            self.sequencer_address.0.0.into(),
            self.block_timestamp.into(),
            self.transaction_count.into(),
            self.event_count.into(),
            self.protocol_version.into(),
            Felt252Wrapper::ZERO,
            self.parent_block_hash.into(),
        ];

        PedersenHasher::compute_hash_on_wrappers(data)
    }
```

Fix everything that breaks due to the removal of the generic parameter

**Comment by elielnfinic** [comment]
Can I be assigned this task?

**Comment by elielnfinic** [comment]
I think this should be closed as PR is merged.

**Comment by tdelabro** [comment]
closed by #1572
